pipelines/predicting.py

```diff
@@ -1,8 +1,9 @@
 from kfp import components, dsl
 from kubernetes.client import V1Toleration
 
-from . import split_bootleg_merge_step, train_step
+from . import split_bootleg_merge_step
 from .common import *
+import pipelines.training
```

Comment:
You should use relative imports instead of absolute ones. Also, why do you need this import? It creates a circular import.

Reply:
The original predicting.py has already imported train_step:
`from . import split_bootleg_merge_step, train_step`,
and it would cause circular import if I use prediction_step in training.py.
That's why I use absolute import to resolve the circular import.

Reply:
Why is absolute import different in this respect?

Reply:
Yeah, using absolute import is a standard trick to solve the circular import.
In short, relative imports will invoke the function, but absolute imports won't.

See here for the explanation of python import mechanisms:
https://agilno.com/why-cyclic-dependency-errors-occur-a-look-into-the-python-import-mechanism/

Reply:
Yeah, what you want instead is:
```
from . import pipelines
```

That's a relative import but also imports the module instead of the function, and should solve your problem.

Reply:
That's good. Addressed in the update.